client-python/pycti/utils/opencti_stix2.py

```diff
@@ -2823,242 +2823,228 @@ def import_item(
+                            self.import_object(item, update, types)
+                    elif (
+                        self.opencti.get_attribute_in_extension("location_type", item)
+                        is not None
+                    ):
+                        if (
                             self.opencti.get_attribute_in_extension(
                                 "location_type", item
-                            )
-                            is not None
+                            ).lower()
+                            in types
                         ):
-                            if (
-                                self.opencti.get_attribute_in_extension(
-                                    "location_type", item
-                                ).lower()
-                                in types
-                            ):
-                                self.import_object(item, update, types)
-            if work_id is not None:
-                self.opencti.work.report_expectation(work_id, None)
-            bundles_success_counter.add(1)
-            return True
-        except (RequestException, Timeout):
-            bundles_timeout_error_counter.add(1)
-            worker_logger.warning("A connection error or timeout occurred")
-            # Platform is under heavy load: wait for unlock & retry almost indefinitely.
-            sleep_jitter = round(random.uniform(10, 30), 2)
-            time.sleep(sleep_jitter)
-            return self.import_item(item, update, types, processing_count + 1, work_id)
-        except Exception as ex:  # pylint: disable=broad-except
-            error = str(ex)
-            error_msg = traceback.format_exc()
-            in_retry = processing_count < PROCESSING_COUNT
-            # Platform is under heavy load, wait for unlock & retry indefinitely.
-            if ERROR_TYPE_LOCK in error_msg:
-                bundles_lock_error_counter.add(1)
-                sleep_jitter = round(random.uniform(1, 3), 2)
-                time.sleep(sleep_jitter)
-                return self.import_item(
-                    item, update, types, processing_count + 1, work_id
-                )
-            # Platform detects a missing reference and have to retry
-            elif ERROR_TYPE_MISSING_REFERENCE in error_msg and in_retry:
-                bundles_missing_reference_error_counter.add(1)
-                sleep_jitter = round(random.uniform(1, 3), 2)
+                            self.import_object(item, update, types)
+        if work_id is not None:
+            self.opencti.work.report_expectation(work_id, None)
+        bundles_success_counter.add(1)
+        return True
+
+    def import_item_with_retries(
+        self,
+        item,
+        update: bool = False,
+        types: List = None,
+        work_id: str = None,
+    ):
+        processing_count = 0
+        worker_logger = self.opencti.logger_class("worker")
+        while processing_count <= MAX_PROCESSING_COUNT:
```

Comment:
look like if processing_count > MAX_PROCESSING_COUNT then the message will be silently discarded ?